src/controllers/UserRequests.js

```diff
@@ -4,23 +4,75 @@ Javascript file meant for handling fetch requests to the backend.
 
 Created by Martin Calcaterra on Wednesday November 16th, 2022
 */
+// for base 64 encodings. needs buffer installed
+import {Buffer} from 'buffer';
 
 
 //the type of data sent
 //pretty sure that putting this here allows it to be accessed by both functions below
 //check if this is good coding practice, using a "global" variable i think.
 const headerData = {
   Accept: "application/json",
-  "Content-Type": "application/json;charset=UTF-8",
+  "Content-Type": "application/json;charset=UTF-8"  
 }
-export function sendUserInfo() {
+
+
+
+export async function getBestIntents(){
+
+  // retrieve the apiKey and versionID from their respective textboxes
+  var submittedApiKey = document.getElementById("apiKey").value;
+  var submittedApiVersion = document.getElementById("versionID").value;
+
+
+  // we need to encode the api key and version to make sure they are written in characters that can be sent across HTTP
+  var encodedInfo = Buffer.from(submittedApiKey + ":" + submittedApiVersion).toString('base64');
```

Comment:
Good job on adding more security to the user's information

Reply:
This actually isn't security; it is encoding, not encryption. The purpose of encryption is to protect user data. Encoding is just a way to write a string using different characaters. This does not increase security; it just expresses the string in a specific set of characters that can be sent through HTTP headers. The string is decoded back to the actual information in the back end controller.